packages/react/src/components/sideNavigation/SideNavigation.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { cloneElement, isValidElement, useEffect } from 'react';
+
+// import core base styles
+import 'hds-core';
+
+import styles from './SideNavigation.module.scss';
+import SideNavigationContext from './SideNavigationContext';
+import { FCWithName } from '../../common/types';
+import classNames from '../../utils/classNames';
+import { useMobile } from '../../hooks/useMobile';
+import { Button } from '../button';
+import { IconAngleDown, IconAngleUp } from '../../icons';
+import { MainLevel } from './mainLevel/MainLevel';
+import { SubLevel } from './subLevel/SubLevel';
+import { useTheme } from '../../hooks/useTheme';
+
+export interface SideNavigationCustomTheme {
+  '--side-navigation-background-color'?: string;
+  '--side-navigation-active-indicator-background-color'?: string;
+  '--side-navigation-icon-size'?: string;
+  '--side-navigation-level-border-color'?: string;
+  '--side-navigation-level-border-color-focus'?: string;
+  '--side-navigation-level-border-color-hover'?: string;
+  '--side-navigation-level-background-color'?: string;
+  '--side-navigation-level-background-color-active'?: string;
+  '--side-navigation-level-background-color-hover'?: string;
+  '--side-navigation-level-color'?: string;
+  '--side-navigation-level-color-active'?: string;
+  '--side-navigation-level-color-hover'?: string;
+  '--side-navigation-mobile-menu-border-color'?: string;
+  '--side-navigation-mobile-menu-z-index'?: number;
+}
+
+export type SideNavigationProps = React.PropsWithChildren<{
+  /**
+   * Additional class names to apply to the side navigation
+   */
+  className?: string;
+  /**
+   * Default value for open main levels
+   */
+  defaultOpenMainLevels?: number[];
+  /**
+   * The id of the side navigation.
+   */
+  id: string;
+  /**
+   * Override or extend the styles applied to the component
+   */
+  style?: React.CSSProperties;
+  /**
+   * Custom theme styles
+   */
+  theme?: SideNavigationCustomTheme;
+  /**
+   * label for the mobile menu toggle button
+   */
+  toggleButtonLabel: string;
+  /**
+   * Autocollapse main levels when opening other main levels.
+   * @default false
+   */
+  autoCollapseOthers?: boolean;
+}>;
+
+export const SideNavigation = ({
+  children,
+  className,
+  defaultOpenMainLevels,
+  id,
+  style,
+  theme,
+  toggleButtonLabel,
+  autoCollapseOthers,
+}: SideNavigationProps) => {
+  const container = React.useRef<HTMLDivElement>(null);
+  const customThemeClass = useTheme<SideNavigationCustomTheme>(styles.sideNavigation, theme);
+  const menuId = `${id}-menu`;
+  const [openMainLevels, setOpenMainLevels] = React.useState<number[]>(defaultOpenMainLevels ?? []);
+  const [mobileMenuOpen, setMobileMenuOpen] = React.useState(false);
+  const isMobile = useMobile();
+
+  const mainLevels = React.Children.map(children, (child, index) => {
+    if (isValidElement(child) && (child.type as FCWithName).componentName === 'MainLevel') {
+      return cloneElement(child, { index });
```

Comment:
Why do we need to give this index to mainLevel component? Could we use the id of the component instead?

Reply:
I think we could, but then the use of property defaultOpenMainLevels might be a bit more complex.
I'll have to test this.